### PR TITLE
Improve change request error handling (#337)

### PR DESCRIPTION
**Pull request by cliffcrosland**
Before, when a client submitted a change request containing an invalid
field, the API would return a 500 Internal Server Error status code, and
invalid ChangeRequest and FieldChange records would remain in the
database afterward.

Now, when a client submits a change request containing an invalid field,
the API returns a 400 Bad Request status along with an error message
describing the incorrect field. Additionally, the entire operation is
wrapped in a Postgres transaction, so invalid ChangeRequest and
FieldChange records inserts are rolled back when validation errors
occur.

### CODE REVIEW

**Comment by lgarofalo** [comment]
This looks like it may just need a `rubocop --auto-correct` to cleanup style.

**Comment by cliffcrosland** [comment]
@lgarofalo Thank you! Submitted another commit, and it appears that Rubocop is happy again.

However, the build appears to be stalling. Seeing this output at the very end of the build log:
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

I read through the build output, and from what I can tell, it appears that all of the tests are passing otherwise.

Perhaps this stalling is caused by the fact that I opened my PR from my own personal fork, `cliffcrosland/askdarcel-api`? I originally tried to a push up this PR branch to the official repo (i.e. `ShelterTechSF/askdarcel-api`), but it appears that I don't have permissions to do so yet 😅 

I think I'll ping Joe and/or Richard to follow up on permissions. Thanks!

**Comment by lgarofalo** [comment]
@cliffcrosland Sounds good, I suspect you may need to be invited to the github organization to push.

**Comment by lgarofalo** [comment]
@cliffcrosland Can you push this from a new remote branch on our repo, not your user fork?

**Comment by cliffcrosland** [comment]
Sounds great! Just pushed my changes to a remote branch here in this repository. Not seeing the option to create a pull request from that branch, but it's probably because this pull request is open. I'll try closing this one first.
